### ISSUE TITLE
ActiveSupport::Duration hours/minutes/seconds is always seconds

### ISSUE THREAD

**Issue by Linuus** [rails/rails#37749]
### Steps to reproduce

```ruby
(8.hours + 1.second).parts
=> {:hours=>8, :seconds=>1}
(8.hours + 1.second).seconds
=> 28801 seconds
(8.hours + 1.second).hours
=> 28801 hours
```

### Expected behavior

I'd expect it do return the number of hours/minutes/seconds.

### Actual behavior

It returns the number of seconds but printed as hours?
Perhaps this is a private API or something, but what it prints is very confusing :)

### System configuration
**Rails version**: 5.2.3

**Ruby version**: 2.6.5

**Comment by prateekkish** [comment]
I've tested this in two environments and both have similar behavior:
1. Rails 5.1.7 Ruby 2.6.5
2. Rails 6.0.0 Ruby 2.6.5

I'll try to work on a PR.

**Comment by pixeltrix** [comment]
So this is a weird effect of this:

```
>> (1.hour).is_a?(Numeric)
=> true
```

Which means that what it's effectively doing is this:

```
>> (1.hour).value.hours
=> 3600 hours
```

So we could define `seconds`, `hours`, etc. on `ActiveSupport::Duration` to return the various parts but would that be the actual parts or the computed value? 🤔 

e.g. which of these would be true:

```
>> (1.hour + 30.minutes).minutes
=> 30
```

or 

```
>> (1.hour + 30.minutes).minutes
=> 90
```

Instinctively I guess you'd probably think the latter but it gets complicated for months, etc. where the period of time it represents varies throughout the year.

**Comment by Linuus** [comment]
There’s an open PR so not really stale.

**Comment by jonathanhefner** [comment]
> So we could define `seconds`, `hours`, etc. on `ActiveSupport::Duration` to return the various parts but would that be the actual parts or the computed value?

> ... but it gets complicated for months, etc. where the period of time it represents varies throughout the year.

Since converting a `Duration` to a single unit is a leaky abstraction, what if the API reflected that?  For example:

```ruby
1.year.estimate_days  # == 365.2425
1.month.estimate_days  # == 30.436875
```

Or, if we prefer integral values, computing an "estimate" also gives license to round (or `ceil` or `floor`) the final result:

```ruby
3.months.estimate_days  # == 91 == (30.436875 * 3).round
```

The `estimate_` prefix also communicates that it is a computed value, and not the named part of the `Duration`.  (When actual part values are required, I think `Duration#parts` is probably sufficient.)

**Comment by rafaelfranca** [comment]
Closed by #38699
